modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -5697,6 +5697,39 @@ public void addOperationToGroup(String tag, String resourcePath, Operation opera
         co.baseName = tag;
     }
 
+    /**
+     * Determines if a constructor call is needed for the parent type of a model `()` 
+     * in the generated code.
+     * 
+     * Important note: This is kotlin specific at the moment, but could be 
+     * extended with to support detection for other languages as well.
+     * 
+     * The generator used to default to issuing a supertype 
+     * constructor call and then it was changed to default to not doing it. 
+     * Both are wrong since there are different cases and the decision has to be
+     * made at runtime based on the parent type itself. 
+     * 
+     * For example if it is a HashMap (because you set 
+     * additionalProperties: true for example) then it MUST HAVE a parent
+     * constructor call. If the parent type is an interface because you are
+     * using allOf with a discriminator property then the generated code for the
+     * model will have it's parent type as the interface which MUST NOT HAVE a
+     * parent constructor call.
+     *  
+     * @see https://github.com/OpenAPITools/openapi-generator/issues/8366
+     * 
+     * @return {boolean} `true` if the () call syntax is needed, false otherwise.
+     */
+    protected boolean isParentCtorCallNeeded(CodegenModel model, String name, Schema schema) {
+        final CodegenProperty property = fromProperty(name, schema, false);
+        // TODO: Right now this is specific to the Kotlin language but it is 
+        // possible or even likely that other languages need the same or similar logic.
+        if (property.baseType == "kotlin.collections.Map") {
+            return true;
+        }
+        return false;
+    }
+
```

Comment:
these logics are only for kotlin. what about moving these to postProcessModels in abstract kotlin generators (or just kotlin spring generator) and store the value in x-is-parent-ctor-call-needed instead?

Reply:
@wing328 Sure, but please bear with me. I've lost the dev env in the meantime so will have to set everything up from scratch for a full test again.

Quick question: Would I also have to move property out of `modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenModel.java` onto some kotlin specific model (if that exists) or is that OK where it is right now?

Reply:
https://github.com/OpenAPITools/openapi-generator/blob/f323a3e788ec932ddc1e31f44f3bd0c8a88f7dbc/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinClientCodegen.java#L876 is a good example where to store it